api/src/main/java/org/ehrbase/api/util/VersionUidHelper.java

```diff
@@ -35,6 +35,19 @@
                             VERSION_PATTERN.toString()
             );
 
+    private UUID uuid;
```

Comment:
@axelsiebert note in the openEHR model the OBJECT_VERSION_ID fields have different types:

uid: UID   // attribute is "uid" not "uuid"
system_id: UID
version_tree_id: VERSION_TREE_ID

UID is abstract, could be UUID, OID or INTERNET_ID

By assigning a UUID/String to the UID fields, and int to the VERSION_TREE_ID is constraining what is possible in the spec. For instance the uid could be "2.16.234.234.4353.364" (an OID), the system_id "org.x.y" (an INTERNET_ID), and the version_tree_id "2.1.3" (a VERSION_TREE_ID).

https://specifications.openehr.org/releases/BASE/latest/base_types.html#_overview_4

Note this class might already be implemented in the Java Ref Impl class and maybe in archie.